LibraBFT/Impl/Consensus/BlockStorage/SyncManager.agda

```diff
@@ -19,7 +19,5 @@ open import Optics.All
 
 module LibraBFT.Impl.Consensus.BlockStorage.SyncManager where
 
-open RWST-do
-
 postulate
-  insertQuorumCertM : QuorumCert → BlockRetriever → LBFT (Either ErrLog Unit)
+  insertQuorumCertM : QuorumCert → BlockRetriever → LBFT (Either FakeErr Unit)
```

Comment:
This was the name @mark-moir introduced. I am also not sure why it is `FakeErr`.

Reply:
I'm sure we discussed this, but for the record...

The point of `FakeErr` is to serve as a placeholder in our model (Agda implementation of the Haskell code) for where the Haskell code logs actual errors, which have a bunch of constructors, most of which take `[Text]` arguments.  We don't want to spend time modeling these errors in detail at this stage (or maybe ever, at least in cases we can prove don't ever happen, e.g., internal invariant checks).  We were initially not modeling the errors at all, but this would affect the "shape" of proofs, so "we" decided to have a placeholder with no structure.  HTH.

Reply:
Perhaps a better way would be to keep `ErrLog`, but just model it with a single constructor `FakeErr`.  That way we could incrementally add real constructors later, perhaps eventually eliminating the`FakeErr` constructor, or spell it differently such as `ImpossibleError`, and then prove that nothing ever emits `ImpossibleError`.  Thoughts?

Reply:
This was the name @mark-moir introduced. I am also not sure why it is `FakeErr`.

Reply:
I don't think there is a need to prove that `ImpossibleError` is never emitted when all the errors have been implemented -- we can just remove that constructor. I think having a datatype for `ErrLog` and *postulating* `fakeErr` (so that no proofs ever match against it) might be a good way to go.

Reply:
My point was that maybe there's not much point modeling the details of errors that we can prove are never emitted, but yes, we could  in principle get to a fully detailed model in time; the maybe-temporary existence of `FakeError` or `ImpossibleError` will enable us to get there incrementally.  Regarding postulating `fakeErr : ErrLog`, I thought that was how I started and you didn't like it :smile:.  Maybe I'm confused though.  In any case, it seems reasonable to me, and this conversation has made me want to make our type signatures look identical again :smile:.